neo4j/instance_store.go

```diff
@@ -0,0 +1,77 @@
+package neo4j
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/pkg/errors"
+
+	bolt "github.com/johnnadratowski/golang-neo4j-bolt-driver"
+)
+
+const addVersionDetailsToInstance = "MATCH (i:`_%s_Instance`) SET i.dataset_id = {dataset_id}, i.edition = {edition}, i.version = {version} RETURN i"
+
+//go:generate moq -out ./mocks/bolt.go -pkg mocks . DBPool BoltConn BoltStmt BoltResult
+
+type BoltConn bolt.Conn
+type BoltStmt bolt.Stmt
+type BoltResult bolt.Result
+
+// DBPool provides a pool of database connections
+type DBPool interface {
+	OpenPool() (bolt.Conn, error)
+	Close() error
+}
+
+type Neo4j struct {
+	Pool DBPool
+}
+
+func (c *Neo4j) AddVersionDetailsToInstance(ctx context.Context, instanceID string, datasetID string, edition string, version int) error {
+	data := log.Data{
+		"instance_id": instanceID,
+		"dataset_id":  datasetID,
+		"edition":     edition,
+		"version":     version,
+	}
+
+	// TODO do we need to do a defensive check first?
+
+	conn, err := c.Pool.OpenPool()
+	if err != nil {
+		return errors.WithMessage(err, "neoClient AddVersionDetailsToInstance: error opening neo4j connection")
+	}
+
+	defer conn.Close()
+
+	query := fmt.Sprintf(addVersionDetailsToInstance, instanceID)
```

Comment:
Would it be safer to add the instance ID to the query like the other fields? Not sure if its actually possible as its in the node type, but may be worth considering if it makes the parameter injection safer.